mutalyzer/website/views.py

```diff
@@ -673,28 +699,80 @@ def check_position(position, field):
 
+      - `reference_accession_number`: RefSeq accession number for the reference
+        sequence.
+      - `sample_accession_number`: RefSeq accession number for the sample
+        sequence.
+    """
     output = Output(__file__)
     output.addMessage(__file__, -1, 'INFO',
                       'Received Description Extract request from %s'
                       % request.remote_addr)
 
+    (r, s, reference_sequence, variant_sequence, reference_accession_number,
+        sample_accession_number, description, raw_vars) = ('', '', '', '', '',
+        '', '', None)
+
+    method = request.args.get('method')
+    if method == 'refseq_method':
+        reference_accession_number = request.args.get(
+            'reference_accession_number') or ''
+        if not reference_accession_number:
+            output.addMessage(__file__, 3, 'EEMPTYFIELD',
+                'Reference accession number input fields is empty.')
+
+        sample_accession_number = request.args.get(
+            'sample_accession_number') or ''
+        if not sample_accession_number:
+            output.addMessage(__file__, 3, 'EEMPTYFIELD',
+                'Sample accession number input fields is empty.')
+
+        if reference_accession_number and sample_accession_number:
+            retriever = Retriever.GenBankRetriever(output)
+            reference_file = retriever.loadrecord(reference_accession_number)
+            sample_file = retriever.loadrecord(sample_accession_number)
+            if reference_file and sample_file:
+                r = unicode(reference_file.seq)
+                s = unicode(sample_file.seq)
+    else: # raw_method
+        reference_sequence = request.args.get('reference_sequence') or ''
+        if not reference_sequence:
+            output.addMessage(__file__, 3, 'EEMPTYFIELD',
+                'Reference sequence number input fields is empty.')
+
+        variant_sequence = request.args.get('variant_sequence') or ''
+        if not variant_sequence:
+            output.addMessage(__file__, 3, 'EEMPTYFIELD',
+                'Sample sequence number input fields is empty.')
+
+        if reference_sequence and variant_sequence:
+            r, s = reference_sequence, variant_sequence
+
     # Todo: Move this to the describe module.
-    if not util.is_dna(reference_sequence):
+    if not util.is_dna(r):
         output.addMessage(__file__, 3, 'ENODNA',
                           'Reference sequence is not DNA.')
-    if not util.is_dna(variant_sequence):
+    if not util.is_dna(s):
         output.addMessage(__file__, 3, 'ENODNA',
                           'Variant sequence is not DNA.')
 
-    raw_vars = extractor.describe_dna(reference_sequence, variant_sequence)
-    description = unicode(raw_vars) #describe.allele_description(raw_vars)
+    if r and s:
+        raw_vars = extractor.describe_dna(r, s)
+        description = unicode(raw_vars) #describe.allele_description(raw_vars)
```

Comment:
Might not be needed (the template can already get it from `raw_vars`), but it might be clearer to keep it (but then there's no need for the template to call the `unicode` filter on it).